Update unique-design-challenges.md with 5) NIPs, and 6) Event deletion

alltheseas opened the pull request:
1) added introductory sectiond design challenge #5 NIPs 
2) added unique event deletion design challenges & open questions

alltheseas committed:
Update unique-design-challenges.md

1) added introductory sectiond design challenge #5 NIPs
2) added unique event deletion design challenges & open questions

alltheseas committed:
Update unique-design-challenges.md

updated terminology from "retract" to "deletion request" per consensus among NIP devs established in https://github.com/nostr-protocol/nips/pull/1425

alltheseas commented:
updated "retract" suggestion to "deletion request" as per NIP dev consensus established in https://github.com/nostr-protocol/nips/pull/1425

alltheseas committed:
Update unique-design-challenges.md

updated oncemore to reflect finalization of NIP-09 terminology to "deletion request"

alltheseas reviewed (commented):
updated to reflect NIP-09 terminology change to "Deletion request"

dtonon commented on docs/unique-design-challenges.md:
```diff
@@ -13,6 +13,7 @@ Nostr, given its nature, presents some unique design challenges and personal rew
 3. Relays
 4. Zaps (if implemented)
 5. NIPs (and why some things are supported by some clients and not others)
+6. Deletion across tens to hundreds of relays (whereas Twitter has a sole database)
```
I would be more generic, without mentioning only Twitter, e.g.:
_whereas closed platforms has a sole centralized database_

dtonon commented on docs/unique-design-challenges.md:
```diff
@@ -76,4 +77,25 @@ While not part of the core nostr protocol, zaps are popular enough that we shoul
+
+## Challenge #5: NIPs
+
+Unlike email clients (e.g. yahoo email client, microsoft outlook email client, gmail email client etc.) which (explaining in an oversimplified way) either support or don't support the email protocol, nostr apps have tens of NIPs, and potentially hundreds of NIP candidates as nostr continues to grow. 
```
I would uppercase the names and avoid the "email client" ripetition:
_(e.g. Yahoo, Microsoft Outlook, Gmail email clients)_

dtonon commented on docs/unique-design-challenges.md:
```diff
@@ -76,4 +77,25 @@ While not part of the core nostr protocol, zaps are popular enough that we shoul
+
+Unlike email clients (e.g. yahoo email client, microsoft outlook email client, gmail email client etc.) which (explaining in an oversimplified way) either support or don't support the email protocol, nostr apps have tens of NIPs, and potentially hundreds of NIP candidates as nostr continues to grow. 
+
+A design consequence of a multitude of opt-in NIPs means that there will be sets of apps with little overlap in supported NIPs. This means that a user of app X may not be able to interact with a user on app Y, or vice versa. This also means that unless this is surfaced to either users, they may not be aware that they cannot interact with users of another app. 
```
I would avoid using X because it could be misunderstood for “Twitter", first of all in this context; A/B seems a better choice.

dtonon commented on docs/unique-design-challenges.md:
```diff
@@ -76,4 +77,25 @@ While not part of the core nostr protocol, zaps are popular enough that we shoul
+
+## Challenge #6: Deletion across tens to hundreds of relays
+
+Event deletion in reference to [NIP-09](https://github.com/nostr-protocol/nips/blob/master/09.md]) on nostr carries a unique challenge of asking various distinct relay (i.e. database) admins to delete a particular event.
```
Without “administrators,” in this way it appears to be a manual action, while the deletion is done by the relays (software) themselves.

alltheseas replied:
Sorry, I don't understand what you are asking with this comment. I trust you to directly modify the statement.

dtonon replied:
I would write:

> Event deletion in reference to [NIP-09](https://github.com/nostr-protocol/nips/blob/master/09.md]) on nostr carries a unique challenge of asking various distinct relays (i.e. databases) to delete a particular event.

Make sense?

alltheseas replied:
Updated

alltheseas committed:
Update unique-design-challenges.md

added nostrability reference to challenge #5, and addressed a couple of @dtonon

alltheseas committed:
Update unique-design-challenges.md

updated wording per @dtonon suggestion to remove admins
